wallet_presentation/src/commonMain/kotlin/net/thechance/mena/wallet/presentation/screen/transaction_history/TransactionHistoryMapper.kt

```diff
@@ -15,39 +15,47 @@ fun Transaction.toUi(): TransactionHistoryScreenState.TransactionHistoryUiState
         id = id,
         timeAndDate = formatLocalDateTime(date = createdAt, outputFormat = "dd MMM, h:mm a"),
         amount = amount.toString(),
-        type = when (type) {
-            TransactionType.SENT -> TransactionHistoryScreenState.TransactionTypeUiState.SENT
-            TransactionType.RECEIVED -> TransactionHistoryScreenState.TransactionTypeUiState.RECEIVED
-            TransactionType.ONLINE_PURCHASE -> TransactionHistoryScreenState.TransactionTypeUiState.ONLINE_SHOPPING
-        },
-        status = when (status) {
-            TransactionStatus.SUCCESS -> TransactionHistoryScreenState.TransactionStatusUiState.SUCCESS
-            TransactionStatus.FAILED -> TransactionHistoryScreenState.TransactionStatusUiState.FAILED
-        },
-        contactName = when (type) {
-            TransactionType.SENT -> receiverName
-            TransactionType.RECEIVED -> senderName
-            else -> null
-        }
+        type = transactionType(),
+        status = transactionStatus(),
+        contactName = userName()
```

Comment:
just change those function names too so that they represent function names more than attributes
```suggestion
        type = getTransactionType(),
        status = getTransactionStatus(),
        contactName = getUserName()
```